File: 30DaysOfCode/day11_2DArrays/2DArrays.cpp
#include <vector>
#include <iostream>

using namespace std;

int main(){
    vector<vector<int>> arr(6,vector<int>(6));
    for(int i = 0; i < 6; i++) {
       for(int j = 0; j < 6; j++) {
          cin >> arr[i][j];
       }
    }
    
    int max = (-9 * 7);
    
    for (int i = 0; i < 4; i++) {
        for (int j = 0; j < 4; j++) {
            int sum = 0;
            sum += (arr[i][j] + arr[i][j+1] + arr[i][j+2]);
            sum += (arr[i+1][j+1]);
            sum += (arr[i+2][j] + arr[i+2][j+1] + arr[i+2][j+2]);
            
            if (sum > max) {
                max = sum;
            }
        }
    }
    
    cout << max << endl;
    
    return 0;
}

File: 30DaysOfCode/day11_2DArrays/2DArrays.pl
#!/usr/bin/perl

$arr_i = 0;
@arr = ();

while($arr_i < 6){
   my $arr_temp = <STDIN>;
   my @arr_t = split / /,$arr_temp;
   chomp @arr_t;
   push @arr,\@arr_t;
   $arr_i++;
}

$max = 0;

for ($i = 0; $i < 4; $i++) {
    for ($j = 0; $j < 4; $j++) {
        $sum = 0;
        
        $sum += $arr[$i][$j] + $arr[$i][$j+1] + $arr[$i][$j+2];
        $sum += $arr[$i+1][$j+1];
        $sum += $arr[$i+2][$j] + $arr[$i+2][$j+1] + $arr[$i+2][$j+2];
        
        if ($sum > $max) {
            $max = $sum;
        }
    }
}

print $max;
